config/api.go

```diff
@@ -1,5 +1,16 @@
 package config
 
+const (
+	ApiTag             = "API"
+	HTTPAuthSecretsTag = "HTTPAuthSecrets"
+)
+
 type API struct {
-	HTTPHeaders map[string][]string // HTTP headers to return with the API.
+	// HTTPHeaders are the HTTP headers to return with the API.
+	HTTPHeaders map[string][]string
+
+	// HTTPAuthSecrets are the secrets used to authenticate in the API.
+	// A secret is in the format "type:value". Check the documentation for
+	// supported types.
+	HTTPAuthSecrets map[string]string `json:",omitempty"`
```

Comment:
I've been thinking about the future where we start leveraging per-user secrets for limiting access to specific commands. `HTTPAuthSecrets` alone will not be enough.

We could keep it as-is, and a place for defining scope of access could be added later as a separate optional `API.AuthScope` (`map[string]*RPCAuthScope` that is expected to use the same keys as `HTTPAuthSecrets`), but it will look hacky as heck.

@hacdias how do you feel about below idea?

Rename it to `API.Authorization`  and make it  `map[string]*RPCAuthScope` where `RPCAuthScope` is a struct similar to `Gateway.PublicGateways`  map, includes secret string but also a place where in future PRs we would add a list of allowed RPC commands and a flag that controls if things like MFS and IPNS are sandboxed per authorized user:
  ```go
  type RPCAuthScope struct {
    Secret string
    AllowedPaths []string // explicit list of RPC path prefixes to allow (none by default, ["/api/v0"] exposes all RPCs)
    Sanboxed Flag // true by default, when set true /api/v0/files|key|name (MFS), /key and /name (IPNS) commands are scoped per Secret, and won't show things that belong to other apps that use own sandbox
  }
  ```

`Sanboxed` is future work but in this PR we could include `AllowedPaths` – validate it inside of the round tripper. This way, MVP provides both HTTP Auth AND ability to limit RPC access to specific commands.

Reply:
I think that sounds sensible, it also allows for more flexibility and future-proofing.